docs/content/usage/authentication.en-us.md

```diff
@@ -349,3 +349,72 @@ If set `ENABLE_REVERSE_PROXY_FULL_NAME=true`, a user full name expected in `X-WE
 You can also limit the reverse proxy's IP address range with `REVERSE_PROXY_TRUSTED_PROXIES` which default value is `127.0.0.0/8,::1/128`. By `REVERSE_PROXY_LIMIT`, you can limit trusted proxies level.
 
 Notice: Reverse Proxy Auth doesn't support the API. You still need an access token or basic auth to make API requests.
+
+## SAML
+
+### Configuring Gitea as a SAML 2.0 Service Provider
+
+- Navigate to `Site Administration > Identity & Access > Authentication Sources`.
+- Click the `Add Authentication Source` button.
+- Select `SAML` as the authentication type.
+
+#### Features Not Yet Supported
+
+Currently, auto-registration is not supported for SAML. During the external account linking process the user will be prompted to set a username and email address or link to an existing account.
+
+SAML group mapping is not supported.
+
+#### Settings
+
+- `Authentication Name` **(required)**
+
+  - The name of this authentication source (appears in the Gitea ACS and metadata URLs)
+
+- `SAML NameID Format` **(required)**
+
+  - This specifies how Identity Provider (IdP) users are mapped to Gitea users. This option will be provider specific.
+
+- `Icon URL` (optional)
+
+  - URL of an icon to display on the Sign-In page for this authentication source.
+
+- `[Insecure] Skip Assertion Signature Validation` (optional)
```

Comment:
Why would this be even a thing? It's like adding an option to not check the user's password in the login form...

Reply:
@techknowlogick Can we remove this?

Reply:
It's correct that it's incredibly insecure, as it bypasses any validation checks for the data that is sent to it, which is via HTTP post so it is accessible to the user, so the user can modify it in transit if validation checks are not added.  It definitely shouldn't be enabled on production systems (after any setup has been done), or even if it does get enabled it should be disabled soon after (even if a system is in testing mode), and disabling access to the application expect to specific machines (likely done via reverse proxy IP allow list) during the time it is enabled.

This was added intentionally, though, for the reason that for debugging purposes in many other systems that support saml provide this as an option. It is to see if the saml is sending the required assertions and the application is parsing it into the correct user fields, and being able to debug that portion prior to seeing if it is a cert error.

Reply:
While I see this feature as useful for development it still should not be an option for the end-user

Reply:
What about marking it in the documentation as
> ⚠️ ⚠️ ⚠️ (experimental) only for debugging purposes and horribly insecure. Might be removed at any time.

Reply:
@lafriks, as the interface is the only way to add/remove SAML auth sources, could you suggest an alternative way to enable/disable this option? What if I add a cron job to disable it after an hour, if it is activated and keep the check box in the interface?

Reply:
> @lafriks, as the interface is the only way to add/remove SAML auth sources, could you suggest an alternative way to enable/disable this option? What if I add a cron job to disable it after an hour, if it is activated and keep the check box in the interface?

Yes, this could be ok but it clearly needs to be stated in the UI accordingly that it will be enabled only temporarily and will be disabled automatically after an hour. Additionally it would be very recommended to show confirmation dialog when this option is enabled on saving auth source to clearly state risks associated with enabling that specific option like:
* Unauthorized user access to server
* User impersonation that could lead to loss of data confidentiality or even loss of data itself.

Reply:
I don't agree with that as it could be forgotten to disable. It should be either that it auto disables or shows clearly visible warning in all pages in UI while it's enabled, to eliminate risk that it is forgotten or is accidentally turned on

Reply:
@lafriks what if instead I add a red banner to every page for admins if it is still enabled? I'm suggesting every page instead of just the admin dashboard as the later would require an admin to view the admin dashboard to be aware of the warning and they may not go their frequently enough.

Reply:
Yes, that would probably be ok in combination with warnings and confirm dialog that states risks when saving auth source

Reply:
I think this shouldn't be in the source code at all. If it stays, why not also have an option saying "Bypass password verification for all local users"?

Reply:
> This was added intentionally, though, for the reason that for debugging purposes in many other systems that support saml provide this as an option. It is to see if the saml is sending the required assertions and the application is parsing it into the correct user fields, and being able to debug that portion prior to seeing if it is a cert error.

It won't help them if the error is so generic that they cannot tell if it's a missing assertion or invalid signature. This should be fixed with proper error handling that says "Assertion rejected, invalid signature" or "Assertion rejected, unable to determine username", not a "allow anyone access to any account on this Gitea instance by checking this one box"

Reply:
I can't speak for all use-cases, but I have found this option (in other systems) to be incredibly helpful in the past. As not having control over both sides of the auth portion, being able to disable the signing portion, had allowed me to let the IAM team know to please check the certs, and it was a quick solve (this sadly happened to me more than once). 99% of the time issues I've run into have been cert related.
I'd disagree that adding a "bypass password check for all users" would be equal, as in that case you would already know if the database had failed, and for ldap the errors would be more specific to what type of tls error is happening.

That being said, I think I'm the only one advocating for the inclusion of this though, and so I'll suggest a poll that'll happen out over the next week (ending the morning of Mar25) to get a gauge from everyone:
😄 include the debug option, but make sure to have the strict language above regarding the warning, including a red banner for admins on each page, and possibly a cronjob to disable to setting after an hour.
🎉 remove the debug option fully

and whatever the results are, I'll follow them.

(if the poll can be updated to use different language, or some other adjustments, please respond below and I'll be happy to update).

Reply:
I guess with the results tied (if we add my vote in) then I will go ahead with the removal.